client/src/components/PaginationController.tsx

```diff
@@ -0,0 +1,87 @@
+import { MouseEventHandler } from 'react';
+import styled from 'styled-components';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { icon } from '@fortawesome/fontawesome-svg-core/import.macro';
+
+interface PaginationControllerProps {
+  paginationRange: number[];
+  currentPage: number;
+  setCurrentPage: Function;
+}
+
+const PaginationController = ({
+  paginationRange,
+  currentPage,
+  setCurrentPage,
+}: PaginationControllerProps) => {
+  // If there are less than 2 times in pagination range we shall not render the component
+  if (currentPage === 0 || paginationRange.length < 2) {
+    return null;
+  }
+
+  const firstPage: number = paginationRange[0];
+  const leftArrowActive: boolean = currentPage !== firstPage;
+  const lastPage: number = paginationRange[paginationRange.length - 1];
+  const rightArrowActive: boolean = currentPage !== lastPage;
+
+  return (
+    <PaginationContainer>
+      <PaginationDiv
+        onClick={() => {
+          if (leftArrowActive) setCurrentPage(currentPage - 1);
+        }}
+      >
+        <FontAwesomeIcon icon={icon({ name: 'chevron-left' })} />
+      </PaginationDiv>
+      {paginationRange?.map((paginationIndex: any, index: number) => {
+        return (
+          <PaginationDiv
+            onClick={() => {
+              if (paginationIndex > -1) {
```

Comment:
can there be any other negative values?
consider making a constant `ELIPSES_DIV = -1` and checking `paginationIndex != ELIPSES_DIV` or `paginatonIndex > ELIPSES_DIV` for readability